src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -617,15 +678,128 @@ private void configureFullTextSearch() {
         Action.TOGGLE_FULL_TEXT_SEARCH,
         actionTextPrefix + " Full Text Search"
       );
-      IconType icon = isFTSEnabled
-        ? IconType.SEARCH_MINUS
-        : IconType.SEARCH_PLUS;
-      actionMenu.setActionIcon(Action.TOGGLE_FULL_TEXT_SEARCH, icon);
     } else {
       actionMenu.setActionVisible(Action.TOGGLE_FULL_TEXT_SEARCH, false);
     }
   }
 
+  private FluentFuture configureFileDownload() {
+    FluentFuture<RestrictionInformationResponse> future = getDoneFuture(null);
+
+    if (entity instanceof FileEntity) {
+      boolean canDownload = entityBundle.getPermissions().getCanDownload();
+      actionMenu.setDownloadMenuEnabled(canDownload);
+      if (canDownload) {
+        actionMenu.setDownloadMenuTooltipText("");
+      } else {
+        String viewOnlyHelpText = authenticationController.isLoggedIn()
+          ? "You don't have download permission. Request access from an administrator, shown under File Tools ➔ File Sharing Settings"
+          : "You need to log in to download this file.";
+        actionMenu.setDownloadMenuTooltipText(viewOnlyHelpText);
+      }
+      actionMenu.setActionVisible(Action.DOWNLOAD_FILE, true);
+      actionMenu.setActionVisible(Action.ADD_TO_DOWNLOAD_CART, true);
+
+      actionMenu.setActionVisible(Action.SHOW_PROGRAMMATIC_OPTIONS, true);
+
+      future = getDoneFuture(entityBundle.getRestrictionInformation());
+      if (entityBundle.getRestrictionInformation() == null) {
+        future =
+          getSynapseJavascriptClient()
+            .getRestrictionInformation(
+              entity.getId(),
+              RestrictableObjectType.ENTITY
+            );
+      }
+      future.addCallback(
+        new FutureCallback<RestrictionInformationResponse>() {
+          @Override
+          public void onSuccess(
+            @NullableDecl RestrictionInformationResponse restrictionInformation
+          ) {
+            ginInjector
+              .getFileDownloadHandlerWidget()
+              .configure(actionMenu, entityBundle, restrictionInformation);
```

Comment:
The FileDownloadHandlerWidget will further configure the actions, as it contains the logic for how to download each type of file handle.